Assets/Scripts/DogAI/DogAI.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * Author(s): Timothy Ng
+ * Description: Brains behind the dogs actions on the world
+ * Usage: [no notes]
+ */
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using k = PPGlobal;
+
+public class DogAI : MonoBehaviour {
+
+    enum State
+    {
+        Idle,
+        Wandering,
+        Eating,
+        Pooping
+    }
+
+    State currentState = State.Wandering;
+    Vector2 wanderCenter;
+    float wanderRadius = 500;
+
+    Vector2 target;
+    int frameCounter = 0;
+
+    int tapCount = 0;
+
+	// Use this for initialization
+	void Start () {
+        wanderCenter = GetComponent<RectTransform>().anchoredPosition;
+        target = GetComponent<RectTransform>().anchoredPosition;
+    }
+	
+	void Update () {
+
+        currentState = DecideState();
+
+        switch (currentState)
+        {
+            case State.Idle:
+                break;
+            case State.Wandering:
+                Wander();
+                break;
+            case State.Eating:
+                break;
+            case State.Pooping:
+                break;
+            default:
+                break;
+        }
+	}
+
+    State DecideState()
+    {
+        if(frameCounter < 120)
+        {
+            frameCounter++;
+            return currentState;
+        }
+        else
+        {
+            frameCounter = 0;
+            switch(Random.Range(0, 2))
+            {
+                case 0:
+                    return State.Idle;
+                case 1:
+                    return State.Wandering;
+            }
+            return State.Idle;
+        }
+
+    }
+
+    void Wander()
+    {
+        if ((target - GetComponent<RectTransform>().anchoredPosition).sqrMagnitude < 0.5f)
+        {
+            float ct = Random.Range(0, 2 * Mathf.PI);
+            float cr = wanderRadius * Mathf.Sqrt(Random.Range(0f, 1f));
+            float x = cr * Mathf.Cos(ct);
+            float y = cr * Mathf.Sin(ct);
+            target = new Vector2(x, y) + wanderCenter;
+        }
+        else
+        {
+            MoveTo(target);
+        }
+
+    }
+
+    void MoveTo(Vector3 target)
```

Comment:
A question to ask of design, but may be better to have set spots / paths / areas which dogs move on. I observed a dog walking up the wall while testing. I'll ask in Slack